packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -43,6 +43,76 @@ const isInvalidFileName = (fileName: string): boolean => {
  * Returns resolved url
  */
 export async function createNewFluidFile(
+    getStorageToken: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    newFileInfo: INewFileInfo,
+    logger: ITelemetryLogger,
+    createNewSummary: ISummaryTree | undefined,
+    epochTracker: EpochTracker,
+): Promise<IOdspResolvedUrl> {
+    return createNewSummary === undefined
+        ? createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker)
+        : createNewFluidFileFromSummary(getStorageToken, newFileInfo, logger, createNewSummary, epochTracker);
+}
+
+export async function createNewEmptyFluidFile(
+    getStorageToken: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    newFileInfo: INewFileInfo,
+    logger: ITelemetryLogger,
+    epochTracker: EpochTracker,
+): Promise<IOdspResolvedUrl> {
+    // Check for valid filename before the request to create file is actually made.
+    if (isInvalidFileName(newFileInfo.filename)) {
+        throwOdspNetworkError("Invalid filename. Please try again.", invalidFileNameStatusCode);
+    }
+
+    const filePath = newFileInfo.filePath ? encodeURIComponent(`/${newFileInfo.filePath}`) : "";
+    // remove .fluid extension for empty files
+    const extensionlessFileName = newFileInfo.filename.endsWith(".fluid")
+        ? newFileInfo.filename.slice(0, newFileInfo.filename.length - 6)
+        : newFileInfo.filename;
+    const encodedFilename = encodeURIComponent(extensionlessFileName);
+    const initialUrl =
+        `${getApiRoot(getOrigin(newFileInfo.siteUrl))}/drives/${newFileInfo.driveId}/items/root:/${filePath
+        }/${encodedFilename}:/content?@name.conflictBehavior=rename&select=id,name,parentReference`;
+
+    const itemId = await getWithRetryForTokenRefresh(async (options) => {
+        const storageToken = await getStorageToken(options, "CreateNewFile");
+
+        return PerformanceEvent.timedExecAsync(
+            logger,
+            { eventName: "createNewFile" },
```

Comment:
I think we can rename it to "createNewEmptyFile" or add a property to the event.